PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/createrediscache/CreateRedisCacheForm.java

```diff
@@ -313,11 +317,19 @@ public void modifyText(ModifyEvent arg0) {
         cbSubs.addSelectionListener(new SelectionAdapter() {
             @Override
             public void widgetSelected(SelectionEvent e) {
-                currentSub = selectedSubscriptions.get(cbSubs.getSelectionIndex());
-                if (loaded) {
-                    fillLocationsAndResourceGrps(currentSub);
+                if (!loaded) {
+                    return;
                 }
-                validateFields();
+                currentSub = selectedSubscriptions.get(cbSubs.getSelectionIndex());
+                AzureTaskManager.getInstance().runOnPooledThread(() -> {
+                    List<Region> locations = com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).listRegions(currentSub.getId());
+                    AzureTaskManager.getInstance().runOnPooledThread(() -> {
```

Comment:
why create another pooled thread to fill items? Besides, do we need to use `runLater` to modify the UI components?